src/staff.cpp

```diff
@@ -142,15 +142,15 @@ int Staff::GetDrawingX() const
 {
     Doc *doc = dynamic_cast<Doc *>(this->GetFirstParent(DOC));
     assert(doc);
-    if (this->HasFacs() && doc->GetType() == Facs) return FacsimileInterface::GetDrawingX();
+    if (this->HasFacs() && (doc->GetType() == Facs)) return FacsimileInterface::GetDrawingX();
```

Comment:
Sorry, I missed that one and in `Staff::GetDrawingY()`. Avoid the dynamic_cast if not necessary.

Reply:
Since `Object::GetFirstParent` will either return `NULL` or a pointer to a `Doc`, would `static_cast` be safe here?

Reply:
It is a good point. Originally, detecting types was always done with dynamic_cast. But this became extremely slow, so the ClassId member was in order to avoid dynamic_cast just for detecting the class of an object. The dynamic_cast were left. Maybe one day we can change them to static_cast, but for now I prefer to remain consistent. Just move it down within the this-HasFacs() condition.